src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ProviderLocationsControllerTests/GetProvidersTrainingLocationsTest.cs

```diff
@@ -55,76 +55,68 @@ public void Before_each_test()
                )))
-               .Returns(verifyUrl)
-               .Callback<UrlRouteContext>(c =>
-               {
-                   verifyRouteValues = c;
-               });
-            _controller.Url = urlHelper.Object;
+               .Returns(BackUrl);
+
+            _urlHelperMock
+               .Setup(m => m.RouteUrl(It.Is<UrlRouteContext>(c =>
+                   c.RouteName.Equals(RouteNames.GetTrainingLocationPostcode)
+               )))
+               .Returns(AddTrainingLocationUrl);
+
+            _controller.Url = _urlHelperMock.Object;
         }
 
         [Test]
         public async Task GetProvidersTrainingLocation_ReturnsValidResponse()
         {
-            var result = await _controller.GetProvidersTrainingLocation();
+            var result = await _controller.GetProvidersTrainingLocations();
 
             var viewResult = result as ViewResult;
             viewResult.Should().NotBeNull();
             viewResult.ViewName.Should().Contain("ViewProviderLocations.cshtml");
             var model = viewResult.Model as ProviderLocationListViewModel;
             model.Should().NotBeNull();
             model.ProviderLocations.Should().NotBeNull();
-            model.BackUrl.Should().Be(verifyUrl);
+            model.BackUrl.Should().Be(BackUrl);
+            model.AddTrainingLocationLink.Should().Be(AddTrainingLocationUrl);
         }
 
         [Test]
         public async Task GetProvidersTrainingLocation_ReturnsNoProviderLocationData()
         {
-            _mediator.Setup(x => x.Send(It.IsAny<GetAllProviderLocationsQuery>(), It.IsAny<CancellationToken>()))
+            _mediatorMock.Setup(x => x.Send(It.IsAny<GetAllProviderLocationsQuery>(), It.IsAny<CancellationToken>()))
                      .ReturnsAsync(() => null);
-            var user = new ClaimsPrincipal(new ClaimsIdentity(new Claim[]{ new Claim(ProviderClaims.ProviderUkprn,"111"),}, "mock"));
-            
-            _controller = new ProviderLocationsController(_mediator.Object, _logger.Object)
-            {
-                ControllerContext = new ControllerContext()
-                {
-                    HttpContext = new DefaultHttpContext() { User = user },
-                },
-                TempData = Mock.Of<ITempDataDictionary>()
-            };
-            _controller.Url = urlHelper.Object;
-            var result = await _controller.GetProvidersTrainingLocation();
+
+            var result = await _controller.GetProvidersTrainingLocations();
 
             var viewResult = result as ViewResult;
             viewResult.Should().NotBeNull();
             viewResult.ViewName.Should().Contain("ViewProviderLocations.cshtml");
             var model = viewResult.Model as ProviderLocationListViewModel;
             model.Should().NotBeNull();
             model.ProviderLocations.Should().BeEmpty();
-            model.BackUrl.Should().Be(verifyUrl);
+            model.BackUrl.Should().Be(BackUrl);
         }
```

Comment:
add test call for AddTrainingLocationLink please

Reply:
Done